packages/mobile/src/components/scrubber/Scrubber.tsx

```diff
@@ -0,0 +1,169 @@
+      flexDirection: 'row',
+      alignItems: 'center'
+    },
+    timestamp: {
+      color: themeColors.neutral,
+      fontSize: 12,
+      flexShrink: 1,
+      width: 25
+    }
+  })
+
+type ScrubberProps = {
+  /**
+   * A unique key to represent this instances of playback.
+   * If the user replays the same track, mediaKey should change
+   */
+  mediaKey: string
+  /**
+   * Whether audio is playing and the scrubber should animate
+   */
+  isPlaying: boolean
+  /**
+   * The duration of the currently playing track
+   */
+  duration: number
+  /**
+   * Callback invoked when focus is gained on the scrubber
+   */
+  onPressIn: () => void
+  /**
+   * Callback invoked when focus is lost on the scrubber
+   */
+  onPressOut: () => void
+}
+
+/**
+ * Scrubber component to control track playback & seek.
+ *
+ * Memoize because the Scrubber is sensitive to parent rerenders
+ * due to its gesture handler use.
+ * useSelectorWeb (used to connect the now playing drawer) induces
+ * rerenders that normally should be harmless.
+ */
+export const Scrubber = memo(
+  ({ mediaKey, isPlaying, duration, onPressIn, onPressOut }: ScrubberProps) => {
+    const styles = useThemedStyles(createStyles)
+    const dispatch = useDispatch()
+
+    // The left-hand timestamp
+    const [timestampStart, setTimestampStart] = useState('')
+    // The right-hand timestamp
+    const [timestampEnd, setTimestampEnd] = useState('')
+
+    const [dragSeconds, setDragSeconds] = useState<string | null>(null)
+
+    const [isDragging, setIsDragging] = useState(false)
+
+    const handlePressIn = useCallback(() => {
+      onPressIn()
+      setIsDragging(true)
+    }, [onPressIn, setIsDragging])
+
+    const handlePressOut = useCallback(
+      percentComplete => {
+        if (global.progress) {
+          if (duration) {
+            setTimestampStart(formatSeconds(percentComplete * duration))
+            dispatch(seek({ type: SEEK, seconds: percentComplete * duration }))
+          }
+        }
+        onPressOut()
+        setIsDragging(false)
+        setTimeout(() => {
+          setDragSeconds(null)
+        }, SCRUB_RELEASE_TIMEOUT_MS)
+      },
+      [onPressOut, setIsDragging, dispatch, duration]
+    )
+
+    // Register an interval to update the start / end timestamps (duration)
+    const seekInterval = useRef<NodeJS.Timeout | null>(null)
+
+    const onDrag = useCallback(
+      percentComplete => {
+        if (global.progress) {
+          if (duration) {
+            setDragSeconds(formatSeconds(percentComplete * duration))
+          }
+        }
+      },
+      [duration, setDragSeconds]
+    )
+
+    useEffect(() => {
+      setDragSeconds(formatSeconds(0))
+      setTimeout(() => {
+        setDragSeconds(null)
+      }, SCRUB_RELEASE_TIMEOUT_MS)
+    }, [mediaKey])
+
+    useEffect(() => {
+      if (seekInterval.current) clearInterval(seekInterval.current)
+      if (!isDragging) {
+        seekInterval.current = setInterval(() => {
```

Comment:
when it comes to setting a ref with an interval, do we just need a cleanup effect at the end here that simply returns clearInterval(seekInterval.current)?